services/coderabbit/coderabbit-stats.service.js

```diff
@@ -0,0 +1,68 @@
+import Joi from 'joi'
+import { BaseJsonService, pathParams } from '../index.js'
+
+const schema = Joi.object({
+  reviews: Joi.number().required(),
+}).required()
+
+class CodeRabbitStats extends BaseJsonService {
+  static category = 'analysis'
+  static route = {
+    base: 'coderabbit',
+    pattern: 'prs/:provider/:org/:repo',
+  }
+
+  static openApi = {
+    '/coderabbit/prs/{provider}/{org}/{repo}': {
+      get: {
+        summary: 'CodeRabbit Pull Request Reviews',
+        description:
+          'This badge pulls the number of PRs reviewed by [CodeRabbit](https://coderabbit.ai), AI code review tool',
+        parameters: pathParams(
+          {
+            name: 'provider',
+            example: 'github, gitlab, bitbucket',
+            description: 'Version Control Provider (e.g., github)',
+          },
+          {
+            name: 'org',
+            example: 'coderabbitai',
+            description: 'Organization or User name',
+          },
+          {
+            name: 'repo',
+            example: 'ast-grep-essentials',
+            description: 'Repository name',
+          },
+        ),
+      },
+    },
+  }
+
+  static defaultBadgeData = {
+    label: 'coderabbit reviews',
+  }
+
+  static render({ reviews }) {
+    return {
+      message: `${reviews}`,
+    }
+  }
+
+  async fetch({ provider, org, repo }) {
+    return this._requestJson({
+      schema,
+      url: `https://api.coderabbit.ai/stats/${provider}/${org}/${repo}`,
+      httpErrors: {
+        404: 'repo not found',
+      },
```

Comment:
You've set a custom value for 404 here, but in the tests you're asserting the the value on the badge is "invalid" rather than "repo not found". The reason for this is that the upstream API actually returns a `400 Bad Request` when the provider, org or repo is not found for the test cases where the org or repo does not exist.

e.g:

```
$ curl "https://api.coderabbit.ai/stats/github/coderabbitai/invalid-repo-name" --include
HTTP/2 400 
...

{"error":"Failed to get CodeRabbit reviews"}
```

It looks like what you really want here is 

```js
httpErrors: {
   400: 'provider or repo not found',
}
```

I can't find any docs on this API. Do you know if there is any case where it actually does return a 404?

Reply:
@chris48s This is a super valid point; I'd ask the team to push the document to our API reference. 

Will update the message to reflect the 400.